projects/plugins/jetpack/modules/header-code.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Header Code.
+ *
+ * Let users insert code snippets into their site header.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Header_Code;
+
+/**
+ * Add code to the head.
+ *
+ * @since $$next-version$$
+ */
+function jetpack_insert_head_code() {
+	$option = get_option( 'jetpack_header_code' );
+
+	// Ensure no code is added on Simple sites.
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		return;
+	}
+
+	if ( ! empty( $option ) && ! is_admin() ) {
+		echo wp_unslash( $option, 'post' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
What's the best way of handling escaping for something like this? I wasn't quite sure.

Reply:
In theory, probably some form of `wp_kses()`. But since the intent of this is to allow for `<script>` tags there doesn't seem much point in bothering, since if script tags are allowed an attacker able to set this option could do anything they want anyway.

Pointing that out in a comment might be helpful.

Reply:
Thanks, will do!